web-common-marlowe/src/Marlowe/Market/Contract5.purs

```diff
@@ -5,197 +5,81 @@ module Marlowe.Market.Contract5
-                                    Close
-                                )
-                                Close
-                            )
-                            ( When
-                                [ Case
-                                    ( Choice
-                                        ( ChoiceId
-                                            "choice"
-                                            (Role "carol")
-                                        )
-                                        [ Bound one one ]
-                                    )
-                                    Close
-                                , Case
-                                    ( Choice
-                                        ( ChoiceId
-                                            "choice"
-                                            (Role "carol")
-                                        )
-                                        [ Bound one one ]
-                                    )
-                                    ( Pay
-                                        (Role "alice")
-                                        (Party (Role "bob"))
-                                        (Token "" "")
-                                        (ConstantParam "amount")
-                                        Close
-                                    )
-                                ]
-                                (SlotParam "arbitrageTimeout")
-                                Close
-                            )
-                        )
-                    ]
-                    (SlotParam "bobTimeout")
-                    ( When
-                        [ Case
-                            ( Choice
-                                ( ChoiceId
-                                    "choice"
-                                    (Role "carol")
-                                )
-                                [ Bound one one ]
-                            )
-                            Close
-                        , Case
-                            ( Choice
-                                ( ChoiceId
-                                    "choice"
-                                    (Role "carol")
-                                )
-                                [ Bound one one ]
-                            )
-                            ( Pay
-                                (Role "alice")
-                                (Party (Role "bob"))
-                                (Token "" "")
-                                (ConstantParam "amount")
-                                Close
-                            )
-                        ]
-                        (SlotParam "arbitrageTimeout")
-                        Close
-                    )
+                ( Deposit
+                    (Role "Dollar provider")
+                    (Role "Dollar provider")
+                    (Token "85bb65" "dollar")
```

Comment:
I thought the first value of a Token construct was the symbol we used to represent the token. For example $... When we show the currency, what should we show in this case? 

Should we have a Map of common tokens to symbols and if it's not found use the second argument as a postfix?

In this case if we didn't have `"85bb65" -> "$"` we could say `1500 dollar`

Reply:
I think it is the hash of the token and the readable symbol is the second field, we can show the hash in a hint maybe. @nau knows better